plugins/destination/gcs/client/spec/schema.go

```diff
@@ -41,7 +41,109 @@ func (s Spec) JSONSchemaExtend(sc *jsonschema.Schema) {
 		},
 	}
 
-	sc.AllOf = append(sc.AllOf, noRotateNoBatch)
+	// path patterns: should be a clean path
+	cleanPath := &jsonschema.Schema{
+		Title: "`path` is a clean path value",
+		Not: &jsonschema.Schema{
+			Title: "`path` is not a clean path value",
+			AnyOf: []*jsonschema.Schema{
+				{
+					Title: "`path` contains `./`",
+					Properties: func() *orderedmap.OrderedMap[string, *jsonschema.Schema] {
+						properties := orderedmap.New[string, *jsonschema.Schema]()
+						properties.Set("path", &jsonschema.Schema{
+							Type:    "string",
+							Pattern: `^.*\./.*$`,
+						})
+						return properties
+					}(),
+				},
+				{
+					Title: "`path` contains `//`",
+					Properties: func() *orderedmap.OrderedMap[string, *jsonschema.Schema] {
+						properties := orderedmap.New[string, *jsonschema.Schema]()
+						properties.Set("path", &jsonschema.Schema{
+							Type:    "string",
+							Pattern: `^.*//.*$`,
+						})
+						return properties
+					}(),
+				},
+			},
+		},
+	}
+
+	pathWithUUID := &jsonschema.Schema{
+		Title: "Require {{UUID}} to be present in path",
+		Properties: func() *orderedmap.OrderedMap[string, *jsonschema.Schema] {
+			// we make the non-zero requirement, so we want to allow only null here
+			properties := orderedmap.New[string, *jsonschema.Schema]()
+			properties.Set("path", &jsonschema.Schema{
+				Type:    "string",
+				Pattern: `^.*\{\{UUID\}\}.*$`,
+			})
+			return properties
+		}(),
+	}
+	// no_rotate:true -> no {{UUID}} should be present in path
+	noRotateNoUUID := &jsonschema.Schema{
+		Title: "Disallow {{UUID}} in path when using no_rotate",
+		If: &jsonschema.Schema{
+			Properties: func() *orderedmap.OrderedMap[string, *jsonschema.Schema] {
+				noRotate := *sc.Properties.Value("no_rotate")
+				noRotate.Default = nil
+				noRotate.Const = true
+				noRotate.Description = ""
+				properties := orderedmap.New[string, *jsonschema.Schema]()
+				properties.Set("no_rotate", &noRotate)
+				return properties
+			}(),
+			Required: []string{"no_rotate"},
+		},
+		Then: &jsonschema.Schema{
+			Not: pathWithUUID,
+		},
+	}
+
+	/* batching enabled -> require {{UUID}} in path or require no path variables in path,
+	since we will use UUID by default if batch */
```

Comment:
Note: Differing behaviour from S3 plugin in order to maintain backwards compatibility for paths without path variables